pkg/util/test/util.go

```diff
@@ -5354,7 +5354,44 @@ func ValidatePodDisruptionBudget(cluster *corev1.StorageCluster, timeout, interv
 
 	// PodDisruptionBudget is supported for k8s version greater than or equal to 1.21 and operator version greater than or equal to 1.5.0
 	// Changing opVersion to 23.10.0 for PTX-23350 | TODO: add better logic with PTX-23407
-	if k8sVersion.GreaterThanOrEqual(minSupportedK8sVersionForPdb) && opVersion.GreaterThanOrEqual(opVer23_10) && opVersion.LessThan(opVer24_2_0) && pxVersion.LessThan(pxVer3_1_2) {
+	// Smart and parallel upgrades is supported from px version 3.1.2 and operator version 24.2.0
+	if k8sVersion.GreaterThanOrEqual(minSupportedK8sVersionForPdb) && opVersion.GreaterThanOrEqual(opVer24_2_0) && pxVersion.GreaterThanOrEqual(pxVer3_1_2) {
+		t := func() (interface{}, bool, error) {
+			nodes, err := operatorops.Instance().ListStorageNodes(cluster.Namespace)
+			if err != nil {
+				return nil, true, fmt.Errorf("failed to get storage nodes, Err: %v", err)
+			}
+			availableNodes := 0
+			for _, node := range nodes.Items {
+				if *node.Status.NodeAttributes.Storage {
+					if node.Status.Phase == "Online" {
+						availableNodes++
+					} else {
+						logrus.Infof("Node %s is in state [%s], PDB might be incorrect", node.Name, node.Status.Phase)
+					}
+				}
+			}
+			pdbs, err := policyops.Instance().ListPodDisruptionBudget(cluster.Namespace)
+			if err != nil {
+				return nil, true, fmt.Errorf("failed to list all poddisruptionbudgets, Err: %v", err)
+			}
+			actualNodePDBCount := 0
+			for _, pdb := range pdbs.Items {
+				if strings.HasPrefix(pdb.Name, "px-") && pdb.Name != "px-kvdb" {
+					actualNodePDBCount++
+				}
+			}
+			if actualNodePDBCount == availableNodes {
```

Comment:
In case of cluster with storageless nodes, this condition will fail right since PDBs will not be created for storageless nodes?

Reply:
Yeah PDB for storageless nodes will not be created, and that is why i only increment availableNodes when storagenode field is true. I had actually tested this with storagelessnodes as well and the test passed